### PR TITLE
Error for removing coin animation

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest
When clicking on the coin animation in the user profile we want to desactivate the coinanimation and by clicking it again it should be activated again, and it should be stored in the database.
Set Coinanimation on true => it worked
Set Coinanimation on false => no update was started on the settings

### Issues
- None

### Todo
- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/UserResolver.ts [hunk 1]
```diff
@@ -537,7 +537,7 @@ export class UserResolver {
     await queryRunner.startTransaction('READ UNCOMMITTED')
 
     try {
-      if (coinanimation) {
+      if (coinanimation !== undefined) {
```
Comment (Mogge) [inline backend/src/graphql/resolver/UserResolver.ts:540]: Why?

Comment (Mogge) [reply backend/src/graphql/resolver/UserResolver.ts:540]:
```
null !== undefined

-> true
```

Comment (Mogge) [reply backend/src/graphql/resolver/UserResolver.ts:540]:
```
if (undefined)
 -> false
```

Comment (Mogge) [reply backend/src/graphql/resolver/UserResolver.ts:540]:
```
if (null)
 -> false
```

Comment (Elweyn) [reply backend/src/graphql/resolver/UserResolver.ts:540]: Yeah cause of: 
```
if (false)
  -> store false
```

Comment (Elweyn) [reply backend/src/graphql/resolver/UserResolver.ts:540]: Now it does not store false, so each time you log back in you get the coin animation, the db update is not triggered on coinanimation === false!

File: backend/src/typeorm/repository/UserSettingRepository.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export class UserSettingRepository extends Repository<UserSetting> {
     let entity = await this.findOne({ userId: userId, key: key })
 
     if (!entity) {
-      entity = new UserSetting()
+      entity = this.create()
```
Comment (Elweyn) [inline backend/src/typeorm/repository/UserSettingRepository.ts:21]:
```suggestion
      entity = new UserSetting()
```

File: backend/src/graphql/resolver/UserResolver.ts (hunk 1 above)
Comment (Elweyn) [reply backend/src/graphql/resolver/UserResolver.ts:540]:
```suggestion
      if (coinanimation !== null && coinanimation !== undefined) {
```

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**